src_kt/java/arcs/crdt/CrdtEntity.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.crdt
+
+import arcs.crdt.internal.Referencable
+
+/**
+ * A [CrdtModel] capable of managing a complex entity consisting of named [CrdtSingleton]s and named
+ * [CrdtSet]s, each of which can manage various types of [Referencable] data.
+ *
+ * ```kotlin
+ * enum class Party {
+ *   Democratic, Republican, Independent
+ * }
+ * data class PersonReference(val id: String)
+ *
+ * CrdtEntity.addFieldValueInterpreters(
```

Comment:
This exact api here is out of date.  This whole sketch serves more as notes for myself at this time, though. It'll get deleted when CrdtEntity is populated.